### ISSUE TITLE
Some issues with Incident Density

### ISSUE THREAD

**Issue by mfunk** [Esri/solutions-geoprocessing-toolbox#266]
Some minor issues with Incident Density:
* Parameter names need to be updated
     * [X] change *Input Features (Incident Points)* to *Input Incident Point Features* for readability
     * [X] move *Bounding Polygons Defining Where Incidents Are Possible* from the parameter name to it's help topic. Name should be something a little more succinct like *Input Incident Boundary Features*.
* [x] Getting an error running tool with default parameters in Pro 1.1 Build : 3100. This problem does NOT happen on ArcMap 10.3.
     * Input Incident Point Features: *All Incidents* (layer from MXD)
     * Input Incident Boundary Features: *District Outline*
     * Output Density Overlay: *C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\id001*
     * Resulting message:

```
Start Time: Mon May 04 13:31:38 2015
Executing (Optimized Hot Spot Analysis): OptimizedHotSpotAnalysis "Incidents\All Incidents" C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\Incidents_OptimizedHotSpotAn # SNAP_NEARBY_INCIDENTS_TO_CREATE_WEIGHTED_POINTS "District Outline" # C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\id001
Start Time: Mon May 04 13:31:41 2015
Running script OptimizedHotSpotAnalysis...
Failed script OptimizedHotSpotAnalysis...
Traceback (most recent call last):
  File "c:\program files\arcgis\pro\Resources\ArcToolbox\Scripts\OptimizedHotSpotAnalysis.py", line 991, in <module>
    setupOptHotSpot()
  File "c:\program files\arcgis\pro\Resources\ArcToolbox\Scripts\OptimizedHotSpotAnalysis.py", line 222, in setupOptHotSpot
    selectionType = UTILS.getSelectionType(featureLayerInit)
  File "c:\program files\arcgis\pro\Resources\ArcToolbox\Scripts\SSUtilities.py", line 155, in getSelectionType
    fidSet = info.FIDSet
AttributeError: DescribeData: Method FIDSet does not exist
Failed to execute (Optimized Hot Spot Analysis).
Failed at Mon May 04 13:31:48 2015 (Elapsed Time: 7.56 seconds)
Failed to execute (IncidentDensity).
Failed at Mon May 04 13:31:48 2015 (Elapsed Time: 10.80 seconds)
```

**Comment by mfunk** [comment]
Tried the same inputs with **Optimized Hot Spot Analysis** in the Spatial Statistics Tools toolbox. I get the same error. So something in ArcGIS Pro is making this fail.

```
arcpy.OptimizedHotSpotAnalysis_stats('Incidents\\All Incidents',"C:\\Workspace\\Test\\TestIncidentAnalysis\\Test_IA\\Test_IA.gdb\\osha_of_004","#", 'SNAP_NEARBY_INCIDENTS_TO_CREATE_WEIGHTED_POINTS', 'District Outline',"#","C:\\Workspace\\Test\\TestIncidentAnalysis\\Test_IA\\Test_IA.gdb\\osha_ds__004")
Runtime error 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "c:\program files\arcgis\pro\Resources\arcpy\arcpy\stats.py", line 791, in OptimizedHotSpotAnalysis
    raise e
  File "c:\program files\arcgis\pro\Resources\arcpy\arcpy\stats.py", line 788, in OptimizedHotSpotAnalysis
    retval = convertArcObjectToPythonObject(gp.OptimizedHotSpotAnalysis_stats(*gp_fixargs((Input_Features, Output_Features, Analysis_Field, Incident_Data_Aggregation_Method, Bounding_Polygons_Defining_Where_Incidents_Are_Possible, Polygons_For_Aggregating_Incidents_Into_Counts, Density_Surface), True)))
  File "c:\program files\arcgis\pro\Resources\arcpy\arcpy\geoprocessing\_base.py", line 498, in <lambda>
    return lambda *args: val(*gp_fixargs(args, True))
arcgisscripting.ExecuteError: 
Traceback (most recent call last):
  File "c:\program files\arcgis\pro\Resources\ArcToolbox\Scripts\OptimizedHotSpotAnalysis.py", line 991, in <module>
    setupOptHotSpot()
  File "c:\program files\arcgis\pro\Resources\ArcToolbox\Scripts\OptimizedHotSpotAnalysis.py", line 222, in setupOptHotSpot
    selectionType = UTILS.getSelectionType(featureLayerInit)
  File "c:\program files\arcgis\pro\Resources\ArcToolbox\Scripts\SSUtilities.py", line 155, in getSelectionType
    fidSet = info.FIDSet
AttributeError: DescribeData: Method FIDSet does not exist

Failed to execute (OptimizedHotSpotAnalysis).
```

@nfeuerstein any idea who works on the Spatial Stats tools, and would they be willing to check this out?

**Comment by nfeuerstein** [comment]
@mfunk - Either of the Laurens (Bennett or Scott) could point you to the right person.

**Comment by conklinbd** [comment]
Is this issue with a daily Pro build? There were issues with some Spatial Stats tools in Pro 1.0 that were fixed in the patch ( I believe).

**Comment by mfunk** [comment]
@conklinbd  ArcGIS Pro 1.1 Build 3100

**Comment by conklinbd** [comment]
Yeah. Hit up the Lauren (s)

**Comment by mfunk** [comment]
@nfeuerstein @conklinbd Closing this one out. I updated to ArcGIS Pro 1.1 Build 3112 and the tool runs just fine:

```
Parameters
Input Incident Point Features	Incidents\All Incidents
Input Incident Boundary Features	District Outline
Output Density Overlay	C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\id_of_001
 
Messages
Start Time: Tue May 05 09:39:13 2015
Executing (Optimized Hot Spot Analysis): OptimizedHotSpotAnalysis "Incidents\All Incidents" C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\Incidents_OptimizedHotSpotAn2 # SNAP_NEARBY_INCIDENTS_TO_CREATE_WEIGHTED_POINTS "District Outline" # C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\id_of_001
Start Time: Tue May 05 09:39:16 2015
Running script OptimizedHotSpotAnalysis...
************************** Initial Data Assessment ***************************
Making sure there are enough incidents for analysis....
- There are 5532 valid input features.
Looking for locational outliers....
- There were 34 outlier locations; these will not be used to compute the snap distance.
Checking to see if the Environment Settings include a raster analysis mask....
- Raster analysis mask not set; constructing convex hull....
**************************** Incident Aggregation ****************************
Snapping nearby incidents together to create weighted points....
- Aggregating features less than 36.7608 Meters from each other.
Evaluating incident counts and the number of weighted point features....
- The aggregation process resulted in 1788 weighted points.
- Incident Count Properties:
Min:        1.0000
        Max:       50.0000
        Mean:       3.0940
        Std. Dev.:  3.9983
***************************** Scale of Analysis ******************************
Looking for an optimal scale of analysis by assessing the intensity of clustering at increasing distances....
- The optimal fixed distance band is based on peak clustering found at 438.2419 Meters
***************************** Hot Spot Analysis ******************************
Finding statistically significant clusters of high and low incident counts....
- There are 91 output features statistically significant based on an FDR correction for multiple testing and spatial dependence.
*********************************** Output ***********************************
Creating output feature class: C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\Incidents_OptimizedHotSpotAn2
- Red output features represent hot spots where high incident counts cluster.
- Blue output features represent cold spots where low incident counts cluster.
Creating output raster layer: C:\Workspace\Test\TestIncidentAnalysis\Test_IA\Test_IA.gdb\id_of_001
- Using optimal fixed distance band (438.2419 Meters) for the kernel density search radius.
- The surface will be clipped to a convex hull of the input points.
Completed script OptimizedHotSpotAnalysis...
Succeeded at Tue May 05 09:39:50 2015 (Elapsed Time: 33.33 seconds)
Succeeded at Tue May 05 09:39:50 2015 (Elapsed Time: 36.61 seconds)
```
